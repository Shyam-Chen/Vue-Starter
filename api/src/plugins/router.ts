import plugin from 'fastify-plugin';

import routes from 'virtual:fastify-routes';

export default plugin(
  async (app) => {
    routes(app, { prefix: '/api' });
  },
  { name: 'router' },
);
