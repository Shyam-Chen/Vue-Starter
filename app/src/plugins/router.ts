import { createRouter, createWebHistory } from 'vue-router';
import nprogress from 'nprogress';

import routes from 'virtual:vue-routes';

const router = createRouter({
  history: createWebHistory(),
  routes: [
    {
      path: '/',
      redirect() {
        if (localStorage.getItem('accessToken')) return { path: '/dashboard' };
        return { path: '/sign-in' };
      },
    },

    ...routes,
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) return { selector: to.hash };
    if (savedPosition) return savedPosition;
    return { top: 0 };
  },
});

router.beforeEach((to, from) => {
  if (to.path !== from.path) nprogress.start();
});

router.afterEach(() => {
  nprogress.done();
});

export default router;
