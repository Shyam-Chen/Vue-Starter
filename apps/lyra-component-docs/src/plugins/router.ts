import { createRouter, createWebHistory } from 'vue-router';

import routes from 'virtual:vue-routes';

const router = createRouter({
  history: createWebHistory(),
  routes: [
    {
      path: '/',
      redirect() {
        if (localStorage.getItem('accessToken')) return { path: '/dashboard' };
        return { path: '/sign-in' };
      },
    },

    ...routes,
  ],
  scrollBehavior(to, from, savedPosition) {
    if (savedPosition) return savedPosition;
    return { top: 0 };
  },
});

export default router;
