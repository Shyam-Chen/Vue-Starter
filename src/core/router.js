import { createWebHistory, createRouter } from 'vue-router';

import Home from '~/Home.vue';

export const history = createWebHistory();

export const router = createRouter({
  history,
  routes: [
    {
      path: '/',
      name: 'home',
      component: Home,
    },
    {
      path: '/hello-world',
      name: 'helloWorld',
      component: () => import('~/modules/hello-world/HelloWorld.vue'),
    },
    {
      path: '/store-pattern',
      name: 'storePattern',
      component: () => import('~/modules/store-pattern/StorePattern.vue'),
    },
    {
      path: '/form-validation',
      name: 'formValidation',
      component: () => import('~/modules/form-validation/FormValidation.vue'),
    },
    {
      path: '/crud-operations',
      component: () => import('~/modules/crud-operations/CrudOperations.vue'),
    },
    {
      path: '/:catchAll(.*)',
      name: 'notFound',
      component: () => import('~/modules/error-handling/NotFound.vue'),
    },
  ],
});
