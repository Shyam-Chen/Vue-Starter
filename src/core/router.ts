import { createWebHistory, createRouter } from 'vue-router';

import Home from './Home.vue';

export const history = createWebHistory();

export const router = createRouter({
  history,
  routes: [
    { path: '/', component: Home },
    { path: '/form-validation', component: () => import('~/modules/form-validation/Registry.vue') },
    {
      path: '/store-pattern',
      component: () => import('~/modules/store-pattern/StorePattern.vue'),
    },
    {
      path: '/:catchAll(.*)',
      component: () => import('~/modules/error-handling/NotFound.vue'),
    },
  ],
});
