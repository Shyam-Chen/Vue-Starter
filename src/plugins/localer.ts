import { createLocaler } from 'vue-localer';

import enUS from '~/locales/en-US';

export default createLocaler({
  fallbackLocale: 'en-US',
  messages: {
    'en-US': enUS,
    ...import.meta.glob(['~/locales/*.ts', '!~/locales/index.ts', '!~/locales/en-US.ts']),
  },
});
