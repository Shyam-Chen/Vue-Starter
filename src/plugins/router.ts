import { createWebHistory, createRouter } from 'vue-router';

import auth from '~/middleware/auth';
import componentsRoutes from '~/modules/components/routes';
import navigationRoutes from '~/modules/navigation/routes';
import formsRoutes from '~/modules/forms/routes';
import chartsRoutes from '~/modules/charts/routes';
import networkRoutes from '~/modules/network/routes';

export const history = createWebHistory();

export const router = createRouter({
  history,
  routes: [
    {
      path: '/',
      redirect(to) {
        if (localStorage.getItem('token')) return { path: '/dashboard' };
        return { path: '/sign-in' };
      },
    },

    {
      path: '/sign-in',
      component: () => import('~/modules/sign-in/Registry.vue'),
      meta: { layout: 'center' },
    },
    {
      path: '/dashboard',
      component: () => import('~/modules/dashboard/Registry.vue'),
      meta: { layout: 'default' },
      beforeEnter: [auth],
    },

    ...componentsRoutes,
    ...navigationRoutes,
    ...formsRoutes,

    ...chartsRoutes,

    ...networkRoutes,

    {
      path: '/:pathMatch(.*)*',
      component: () => import('~/Error.vue'),
      meta: { layout: 'empty' },
    },
  ],
});
