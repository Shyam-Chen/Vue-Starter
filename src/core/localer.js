import { createLocaler } from 'vue-localer';

import enUS from '~/locales/en-US.js';

import { router } from './router.js';

export const langList = {
  'en-US': 'American English',
  'ja-JP': '日本語',
  'zh-TW': '正體中文',
};

export const getUserLang = () => {
  const langListKeys = Object.keys(langList);

  if (langListKeys.includes(navigator.language)) {
    return navigator.language;
  }

  return 'en-US';
};

function camelToHyphen(str) {
  return str.replace(/([A-Z])/g, (match) => '-' + match.toLowerCase());
}

export const localer = createLocaler({
  data: {
    lang: 'en-US',
    locale: enUS,
  },
  methods: {
    async initialLanguage({ commit }, mod) {
      const lang = localStorage.getItem('lang') || getUserLang();
      document.documentElement.lang = lang;
      const res = await import(`../locales/${lang}.js`);
      commit('injectLanguage', { lang, locale: res.default });

      if (mod) {
        const modRes = await import(`../modules/${camelToHyphen(mod)}/locales/${lang}.js`);
        commit(`${mod}/injectLanguage`, { locale: modRes.default });
      }
    },
    async setLanguage({ commit }, lang) {
      document.documentElement.lang = lang;
      localStorage.setItem('lang', lang);
      const res = await import(`../locales/${lang}.js`);
      commit('injectLanguage', { lang, locale: res.default });

      if (router.currentRoute.value.name !== 'home') {
        let mod = router.currentRoute.value.name.split('/')[0];
        const modRes = await import(`../modules/${camelToHyphen(mod)}/locales/${lang}.js`);
        commit(`${mod}/injectLanguage`, { locale: modRes.default });
      }
    },
  },
});
