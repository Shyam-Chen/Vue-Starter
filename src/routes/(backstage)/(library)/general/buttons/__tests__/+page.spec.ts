import { mount } from '@vue/test-utils';

import router from '~/plugins/router';
import localer from '~/plugins/localer';

import Page from '../+page.vue';

test('/general/buttons', async () => {
  const wrapper = mount(Page, { global: { plugins: [router, localer] } });
  expect(wrapper.html()).toMatchSnapshot();
});
