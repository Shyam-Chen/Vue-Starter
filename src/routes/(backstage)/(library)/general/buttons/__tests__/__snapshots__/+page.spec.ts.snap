// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`/general/buttons 1`] = `
"<nav class="rounded-md w-full">
  <ol class="list-reset flex">
    <li><span class="text-slate-500">Library</span></li>
    <li>
      <div class="mx-2 flex items-center h-full">/</div>
    </li>
    <li><span class="text-slate-500">General</span></li>
    <li>
      <div class="mx-2 flex items-center h-full">/</div>
    </li>
    <li><span class="text-slate-500">Button</span></li>
    <!--v-if-->
  </ol>
</nav>
<h1 class="text-4xl font-extrabold my-4">Button</h1>
<section class="my-8">
  <h2 class="text-3xl font-bold my-4">Basic</h2>
  <div data-v-ac05e22a="" class="Card">
    <div data-v-ac05e22a="" class="CardHeader"></div>
    <div data-v-ac05e22a="" class="CardBody">
      <div data-v-ac05e22a="" class="CardTitle"></div>
      <div data-v-ac05e22a="" class="CardSubtitle"></div>
      <div data-v-ac05e22a="" class="CardContent">
        <div class="space-y-4">
          <div class="flex flex-col md:flex-row gap-4"><button data-v-54761eda="" type="button" class="Button contained primary small">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Contained</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined primary small">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Outlined</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button text primary small">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Text</div>
              <!--v-if-->
            </button></div>
          <div class="flex flex-col md:flex-row gap-4"><button data-v-54761eda="" type="button" class="Button contained primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Contained</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Outlined</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button text primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Text</div>
              <!--v-if-->
            </button></div>
          <div class="flex flex-col md:flex-row gap-4"><button data-v-54761eda="" type="button" class="Button contained primary large">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Contained</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined primary large">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Outlined</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button text primary large">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Text</div>
              <!--v-if-->
            </button></div>
        </div>
      </div>
      <div data-v-ac05e22a="" class="CardFooter"></div>
    </div>
  </div>
</section>
<section class="my-8">
  <h2 class="text-3xl font-bold my-4 pt-6">Icons</h2>
  <div data-v-ac05e22a="" class="Card">
    <div data-v-ac05e22a="" class="CardHeader"></div>
    <div data-v-ac05e22a="" class="CardBody">
      <div data-v-ac05e22a="" class="CardTitle"></div>
      <div data-v-ac05e22a="" class="CardSubtitle"></div>
      <div data-v-ac05e22a="" class="CardContent">
        <div class="space-y-4">
          <div class="flex flex-col md:flex-row gap-4"><button data-v-54761eda="" type="button" class="Button contained primary small icon">
              <div data-v-54761eda="" class="Button-Icon i-material-symbols-add-rounded small"></div>
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained primary small">
              <!--v-if-->
              <div data-v-54761eda="" class="Button-Icon flex items-center overflow-hidden small">
                <div data-v-54761eda="" class="Button-Icon i-material-symbols-search-rounded small"></div>
              </div>
              <div data-v-54761eda="">Search</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained primary small">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Next</div>
              <div data-v-54761eda="" class="Button-Icon flex items-center overflow-hidden small">
                <div data-v-54761eda="" class="Button-Icon i-material-symbols-arrow-right-alt-rounded small"></div>
              </div>
            </button></div>
          <div class="flex flex-col md:flex-row gap-4"><button data-v-54761eda="" type="button" class="Button contained primary icon">
              <div data-v-54761eda="" class="Button-Icon i-material-symbols-add-rounded"></div>
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained primary">
              <!--v-if-->
              <div data-v-54761eda="" class="Button-Icon flex items-center overflow-hidden">
                <div data-v-54761eda="" class="Button-Icon i-material-symbols-search-rounded"></div>
              </div>
              <div data-v-54761eda="">Search</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Next</div>
              <div data-v-54761eda="" class="Button-Icon flex items-center overflow-hidden">
                <div data-v-54761eda="" class="Button-Icon i-material-symbols-arrow-right-alt-rounded"></div>
              </div>
            </button></div>
          <div class="flex flex-col md:flex-row gap-4"><button data-v-54761eda="" type="button" class="Button contained primary large icon">
              <div data-v-54761eda="" class="Button-Icon i-material-symbols-add-rounded large"></div>
              <!--v-if-->
              <!--v-if-->
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained primary large">
              <!--v-if-->
              <div data-v-54761eda="" class="Button-Icon flex items-center overflow-hidden large">
                <div data-v-54761eda="" class="Button-Icon i-material-symbols-search-rounded large"></div>
              </div>
              <div data-v-54761eda="">Search</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained primary large">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Next</div>
              <div data-v-54761eda="" class="Button-Icon flex items-center overflow-hidden large">
                <div data-v-54761eda="" class="Button-Icon i-material-symbols-arrow-right-alt-rounded large"></div>
              </div>
            </button></div>
        </div>
      </div>
      <div data-v-ac05e22a="" class="CardFooter"></div>
    </div>
  </div>
</section>
<section class="my-8">
  <h2 class="text-3xl font-bold my-4 pt-6">Disabled</h2>
  <div data-v-ac05e22a="" class="Card">
    <div data-v-ac05e22a="" class="CardHeader"></div>
    <div data-v-ac05e22a="" class="CardBody">
      <div data-v-ac05e22a="" class="CardTitle"></div>
      <div data-v-ac05e22a="" class="CardSubtitle"></div>
      <div data-v-ac05e22a="" class="CardContent">
        <div class="flex flex-col md:flex-row gap-4"><button data-v-54761eda="" type="button" class="Button contained primary disabled" disabled="">
            <!--v-if-->
            <!--v-if-->
            <div data-v-54761eda="">Contained</div>
            <!--v-if-->
          </button><button data-v-54761eda="" type="button" class="Button outlined primary disabled" disabled="">
            <!--v-if-->
            <!--v-if-->
            <div data-v-54761eda="">Outlined</div>
            <!--v-if-->
          </button><button data-v-54761eda="" type="button" class="Button text primary disabled" disabled="">
            <!--v-if-->
            <!--v-if-->
            <div data-v-54761eda="">Text</div>
            <!--v-if-->
          </button></div>
      </div>
      <div data-v-ac05e22a="" class="CardFooter"></div>
    </div>
  </div>
</section>
<section class="my-8">
  <h2 class="text-3xl font-bold my-4 pt-6">Loading</h2>
  <div data-v-ac05e22a="" class="Card">
    <div data-v-ac05e22a="" class="CardHeader"></div>
    <div data-v-ac05e22a="" class="CardBody">
      <div data-v-ac05e22a="" class="CardTitle"></div>
      <div data-v-ac05e22a="" class="CardSubtitle"></div>
      <div data-v-ac05e22a="" class="CardContent">
        <div class="space-y-4"><button data-v-54761eda="" type="button" class="Button contained primary small">
            <!--v-if-->
            <div data-v-54761eda="" class="Button-Icon flex items-center overflow-hidden small">
              <div data-v-54761eda="" class="Button-Icon i-material-symbols-search-rounded small"></div>
            </div>
            <div data-v-54761eda="">Search</div>
            <!--v-if-->
          </button><button data-v-54761eda="" type="button" class="Button contained primary">
            <!--v-if-->
            <div data-v-54761eda="" class="Button-Icon flex items-center overflow-hidden">
              <div data-v-54761eda="" class="Button-Icon i-material-symbols-search-rounded"></div>
            </div>
            <div data-v-54761eda="">Search</div>
            <!--v-if-->
          </button><button data-v-54761eda="" type="button" class="Button contained primary large">
            <!--v-if-->
            <div data-v-54761eda="" class="Button-Icon flex items-center overflow-hidden large">
              <div data-v-54761eda="" class="Button-Icon i-material-symbols-search-rounded large"></div>
            </div>
            <div data-v-54761eda="">Search</div>
            <!--v-if-->
          </button></div>
      </div>
      <div data-v-ac05e22a="" class="CardFooter"></div>
    </div>
  </div>
</section>
<section class="my-8">
  <h2 class="text-3xl font-bold my-4 pt-6">Colors</h2>
  <div data-v-ac05e22a="" class="Card">
    <div data-v-ac05e22a="" class="CardHeader"></div>
    <div data-v-ac05e22a="" class="CardBody">
      <div data-v-ac05e22a="" class="CardTitle"></div>
      <div data-v-ac05e22a="" class="CardSubtitle"></div>
      <div data-v-ac05e22a="" class="CardContent">
        <div class="space-y-4">
          <div class="flex flex-col md:flex-row flex-wrap gap-4"><button data-v-54761eda="" type="button" class="Button contained primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Primary</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained secondary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Secondary</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained success">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Success</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained danger">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Danger</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained warning">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Warning</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button contained info">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Info</div>
              <!--v-if-->
            </button></div>
          <div class="flex flex-col md:flex-row flex-wrap gap-4"><button data-v-54761eda="" type="button" class="Button outlined primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Primary</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined secondary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Secondary</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined success">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Success</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined danger">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Danger</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined warning">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Warning</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined info">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Info</div>
              <!--v-if-->
            </button></div>
          <div class="flex flex-col md:flex-row flex-wrap gap-4"><button data-v-54761eda="" type="button" class="Button text primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Primary</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button text secondary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Secondary</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button text success">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Success</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button text danger">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Danger</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button text warning">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Warning</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button text info">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Info</div>
              <!--v-if-->
            </button></div>
        </div>
      </div>
      <div data-v-ac05e22a="" class="CardFooter"></div>
    </div>
  </div>
</section>
<section class="my-8">
  <h2 class="text-3xl font-bold my-4 pt-6">Template</h2>
  <div data-v-ac05e22a="" class="Card">
    <div data-v-ac05e22a="" class="CardHeader"></div>
    <div data-v-ac05e22a="" class="CardBody">
      <div data-v-ac05e22a="" class="CardTitle"></div>
      <div data-v-ac05e22a="" class="CardSubtitle"></div>
      <div data-v-ac05e22a="" class="CardContent"><button data-v-54761eda="" type="button" class="Button outlined primary"> Template <div data-v-cb727b1e="" class="flex">
            <div data-v-cb727b1e="" class="Chip primary">5
              <!--v-if-->
            </div>
          </div></button></div>
      <div data-v-ac05e22a="" class="CardFooter"></div>
    </div>
  </div>
</section>
<section class="my-8">
  <h2 class="text-3xl font-bold my-4 pt-6">Button Group</h2>
  <div data-v-ac05e22a="" class="Card">
    <div data-v-ac05e22a="" class="CardHeader"></div>
    <div data-v-ac05e22a="" class="CardBody">
      <div data-v-ac05e22a="" class="CardTitle"></div>
      <div data-v-ac05e22a="" class="CardSubtitle"></div>
      <div data-v-ac05e22a="" class="CardContent">
        <div data-v-496d006f="" class="ButtonGroup"><button data-v-54761eda="" type="button" class="Button contained primary">
            <!--v-if-->
            <!--v-if-->
            <div data-v-54761eda="">Year</div>
            <!--v-if-->
          </button><button data-v-54761eda="" type="button" class="Button contained primary">
            <!--v-if-->
            <!--v-if-->
            <div data-v-54761eda="">Month</div>
            <!--v-if-->
          </button><button data-v-54761eda="" type="button" class="Button contained primary">
            <!--v-if-->
            <!--v-if-->
            <div data-v-54761eda="">Week</div>
            <!--v-if-->
          </button><button data-v-54761eda="" type="button" class="Button contained primary">
            <!--v-if-->
            <!--v-if-->
            <div data-v-54761eda="">Day</div>
            <!--v-if-->
          </button></div>
      </div>
      <div data-v-ac05e22a="" class="CardFooter"></div>
    </div>
  </div>
  <section class="my-4">
    <h3 class="text-2xl font-semibold my-4 pt-2">Controlled</h3>
    <div data-v-ac05e22a="" class="Card">
      <div data-v-ac05e22a="" class="CardHeader"></div>
      <div data-v-ac05e22a="" class="CardBody">
        <div data-v-ac05e22a="" class="CardTitle"></div>
        <div data-v-ac05e22a="" class="CardSubtitle"></div>
        <div data-v-ac05e22a="" class="CardContent">
          <div data-v-496d006f="" class="ButtonGroup"><button data-v-54761eda="" type="button" class="Button outlined primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Year</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Month</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Week</div>
              <!--v-if-->
            </button><button data-v-54761eda="" type="button" class="Button outlined primary">
              <!--v-if-->
              <!--v-if-->
              <div data-v-54761eda="">Day</div>
              <!--v-if-->
            </button></div>
        </div>
        <div data-v-ac05e22a="" class="CardFooter"></div>
      </div>
    </div>
  </section>
</section>"
`;
